controllers/oauth.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+const ClientOAuth2 = require("client-oauth2");
+const crypto = require("crypto");
+const got = require("got");
+
+const AppConstants = require("../app-constants");
+const DB = require("../db/DB");
+
+
+// This object exists instead of inlining the env vars to make it easy
+// to abstract fetching API endpoints from the OAuth server (instead
+// of specifying them in the environment) in the future.
+const FxAOAuthUtils = {
+  get authorizationUri() { return AppConstants.OAUTH_AUTHORIZATION_URI; },
+  get tokenUri() { return AppConstants.OAUTH_TOKEN_URI; },
+  get profileUri() { return AppConstants.OAUTH_PROFILE_URI; },
+};
+
+const FxAOAuth = new ClientOAuth2({
```

Comment:
Small nit, but should this be called `FxAOAuthClient` ("Utils" vs "Client"), plus then below it may be a bit clearer as `function init(req, res, client = FxAOAuthClient) {...}`.